tests/UKHO.S100PermitService.API.UnitTests/Controller/PermitControllerTests.cs

```diff
@@ -69,20 +70,56 @@ public async Task WhenPermitGeneratedSuccessfully_ThenReturnsZipStreamResponse()
         }
 
         [Test]
-        public async Task WhenPermitGenerationFailed_ThenReturnsNoContentResponse()
+        [TestCase(HttpStatusCode.BadRequest)]
+        [TestCase(HttpStatusCode.NotFound)]
+        [TestCase(HttpStatusCode.NoContent)]
+        [TestCase(HttpStatusCode.InternalServerError)]
+        public async Task WhenPermitGenerationFailed_ThenReturnsNotOkResponse(HttpStatusCode httpStatusCode)
         {
-            A.CallTo(() => _fakePermitService.ProcessPermitRequestAsync(A<int>.Ignored, A<CancellationToken>.Ignored, A<string>.Ignored))
-                .Returns((HttpStatusCode.NoContent, new MemoryStream()));
-
-            var result = (ObjectResult)await _permitController.GeneratePermits(1);
-
-            result.StatusCode.Should().Be(StatusCodes.Status204NoContent);
+            A.CallTo(() => _fakePermitService.ProcessPermitRequestAsync(A<int>.Ignored, A<string>.Ignored, A<CancellationToken>.Ignored))
+                .Returns(GetPermitServiceResult(httpStatusCode));
+
+            var result = await _permitController.GeneratePermits(1);
+
+            switch(result)
+            {
+                case StatusCodeResult statusCodeResult when statusCodeResult.StatusCode == (int)httpStatusCode: // 500: InternalServerError
+                    statusCodeResult.StatusCode.Should().Be((int)httpStatusCode);
+                    break;
+
+                case BadRequestObjectResult badRequestObjectResult when badRequestObjectResult.StatusCode == (int)httpStatusCode: //400: BadRequest
+                    badRequestObjectResult.StatusCode.Should().Be((int)httpStatusCode);
+                    badRequestObjectResult.Value.Should().BeEquivalentTo(new
+                    {
+                        Errors = new List<ErrorDetail>
+                    {
+                        new() { Description = "LicenceId is incorrect", Source = "GetUserPermits" }
+                    }
+                    });
+                    break;
+
+                case NotFoundObjectResult notFoundObjectResult when notFoundObjectResult.StatusCode == (int)httpStatusCode: //404: NotFound
+                    notFoundObjectResult.StatusCode.Should().Be((int)httpStatusCode);
+                    notFoundObjectResult.Value.Should().BeEquivalentTo(new
+                    {
+                        Errors = new List<ErrorDetail>
+                    {
+                        new() { Description = "Licence Not Found", Source = "GetUserPermits" }
```

Comment:
Please check in other Unit Test classes as well and make below changes
```suggestion
                        new() { Description = "Licence not found", Source = "licenceId" }
```

Reply:
Resolved